sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2118,6 +2130,27 @@ std::pair<RT::PiKernel, std::mutex *> ProgramManager::getOrCreateKernel(
                         &(BuildResult->MBuildResultMutex));
 }
 
+bool doesDevSupportImgAspects(const device &Dev,
+                              const RTDeviceBinaryImage &Img) {
+  const RTDeviceBinaryImage::PropertyRange &PropRange =
+      Img.getDeviceRequirements();
+  for (RTDeviceBinaryImage::PropertyRange::ConstIterator Prop : PropRange) {
+    using namespace std::literals;
+    if ((*Prop)->Name == "aspects"sv) {
```

Comment:
nit: IMO, idiomatic way is

```
auto AspectIt = std::find_if();
if (AspectIt == PropRange.end())
  return true;
// Rest of the checks
```

Reply:
Also thought of this, so [904c39b](https://github.com/intel/llvm/pull/7691/commits/904c39bf08a98c8f6e549b3b8a691c6258a8f295)